test/Middleware/ImplicitOptionsMiddlewareTest.php

```diff
@@ -111,4 +108,21 @@ public function testInjectsAllowHeaderInResponseProvidedToConstructorDuringOptio
         $result = $this->middleware->process($request->reveal(), $handler->reveal());
         $this->assertSame($this->response->reveal(), $result);
     }
+
+    public function testReturnsResultOfHandlerWhenRouteNotFound()
+    {
+        $result = RouteResult::fromRouteFailure(Route::HTTP_METHOD_ANY);
```

Comment:
This is the main test, so in case route NOT FOUND we have allowed any methods, but the result is failure.
So `isFailure === true` and `isMethodFailure === false`. Then -> 404.